injectproxy/routes.go

```diff
@@ -291,7 +307,7 @@ func (r *routes) query(w http.ResponseWriter, req *http.Request) {
 	e := NewEnforcer(r.errorOnReplace,
 		[]*labels.Matcher{{
 			Name:  r.label,
-			Type:  labels.MatchEqual,
+			Type:  labels.MatchRegexp,
```

Comment:
Do we need to do any special treatment of the labels so that someone doesn't intentionally insert a regular expression to DOS the proxy, e.g. with lots of backtracking?

Reply:
@squat Do you have any example regex expression which could lead to a DOS for testing?

Reply:
Shouldn't we use [QuoteMeta](https://pkg.go.dev/regexp#QuoteMeta) to avoid any risk like this?

Reply:
Yes exactly, that's perfect @simonpasquier this avoids the issue i was worried about

Reply:
> Shouldn't we use [QuoteMeta](https://pkg.go.dev/regexp#QuoteMeta) to avoid any risk like this?

Thank you for your feedback. I added a QuoteMeta in this line according to your suggestion. I hope I understand you correctly.

https://github.com/prometheus-community/prom-label-proxy/pull/115/files#diff-d120278b267943f472f753843c2abce05ac45a07a1e53b512dfff1438125b980R284